src/libcrun/linux.c

```diff
@@ -3192,11 +3192,22 @@ validate_sysctl (const char *original_value, const char *name, unsigned long nam
 
       if (strcmp (name, "kernel/domainname") == 0)
         {
-          if (namespaces_created & CLONE_NEWUTS)
-            return 0;
+          if (! is_empty_string (def->domainname))
+            {
+              // Value of sysctl `kernel/domainname` is going to
+              // conflict with already set field `domainname` in
+              // OCI spec, in such scenario crun will fail to prevent
+              // unexpected behaviour for end user.
+              return crun_make_error (err, 0, "the sysctl `%s` conflicts with OCI field `domainname`", original_value);
```

Comment:
Probably no need to raise an error if `strcmp(name, def->domainname)  == 0`

Reply:
@AkihiroSuda I think you meant `strcmp(original_value, def->domainname)` , since `name` is the key of `sysctl`  which is `kernel.domainname`.  

I have added this condition and also a test for this.